content/en/docs/configuration/selfsigned.md

```diff
@@ -58,3 +52,47 @@ selfsigned-issuer   True                          2m
 
 Certificates are now ready to be requested by using the `SelfSigned` `Issuer`
 named `selfsigned-issuer` within the `sandbox` namespace.
+
+### CRL Distribution Points
+
+You may also optionally specify [CRL](https://en.wikipedia.org/wiki/Certificate_revocation_list)
+Distribution Points as an array of strings, each of which identifies the location of a CRL in
+which the revocation status of issued certificates can be checked:
+
+```yaml
+...
+spec:
+  selfSigned:
+    crlDistributionPoints:
+    - "http://example.com"
+```
+
+
+## Caveats
+
+### Trust
+
+Clients consuming `SelfSigned` certificates have _no way_ to trust them
+without already having the certificates beforehand. As such, most clients are
+likely to be forced to "TOFU" (trust on first use), which has security
+implications in the event of a man-in-the-middle attack.
+
+### Certificate Validity
+
+One side-effect of a certificate being self-signed is that its Subject DN and
+its Issuer DN are identical. The X.509[RFC 5280, section 4.1.2.4](https://tools.ietf.org/html/rfc5280#section-4.1.2.4) requires that:
+
+> The issuer field MUST contain a non-empty distinguished name (DN).
+
+However, self-signed certs don't have a subject DN set by default. Unless you
+manually set a Subject, the Issuer will be empty and the certificate will
+technically be invalid.
+
+Validation of this specific area of the spec is patchy and varies between TLS
+libraries, but there's always the risk that a library will improve its
+validation - entirely within spec - in the future and break your app if you're
+using a certificate with an empty Issuer DN.
+
+To avoid this, be sure to set a Subject for `SelfSigned` certs. In any case,
+cert-manager will emit a Kubernetes warning event if it detects that an invalid
```

Comment:
Suggestion: maybe we could specify what event or perhaps just point at the code that does that https://github.com/jetstack/cert-manager/blob/master/pkg/controller/certificaterequests/selfsigned/selfsigned.go#L140? Not sure if that's needed, but I guess I would be curious what event/log message to look for.